### ISSUE TITLE
Don't enforce trailing spaces on newlines (typedef-whitespace)

### ISSUE THREAD

**Issue by SteveVanOpstal** [palantir/tslint#1354]
### Bug Report

- __TSLint version__: `3.12.1`
- __TypeScript version__: `1.9.0-dev.20160610-1.0`
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
public aVeryLongFunctionThatExceeds100Characters(parameter1: string, parameter2: string):
    TheReturnType {
  ...
}
```

with `tslint.json` configuration:

```json
"rules": {
    "no-trailing-whitespace": true,
    "max-line-length": [ true, 100 ],
    "typedef-whitespace": [
      true,
      {},
      { "call-signature": "space" }
    ],
}
```

#### Actual behavior

When there is no space after the colon:

> expected space after colon in call-signature

When there is a space after the colon:

> trailing whitespace

#### Expected behavior

The errors are unexpected. Line breaks should be excluded from the whitespace-checking.
Although the checks do what they need to (check for a whitespaces) it is unexpected behavior.

This is opinionated so I would like to know if someone has another view on the matter or has a situation where whitespaces and newlines are useful.

**Comment by adidahiya** [comment]
I generally agree with you -- a trailing whitespace should not be enforced by the `typedef-whitespace` rule when there is a newline present at that location.
